src/lib/client.ts

```diff
@@ -190,28 +190,9 @@ export class Client {
       ((await invoke("plot_progress_tracker")) as number) / 256
     return plot_progress_tracker <= 1 ? 1 : plot_progress_tracker - 1
   }
-  // Used to check archived segments locally and display farmer plot status.
-  // TODO: Improve this, check if can get from storage queries.
-  public async getNetworkSegmentIndex(hash?: Hash): Promise<number> {
-    let signedBlock
-    if (hash) signedBlock = await this.publicApi.rpc.chain.getBlock(hash)
-    else signedBlock = await this.publicApi.rpc.chain.getBlock()
-    if (signedBlock.block.header.number.toNumber() < 100) return 1
-    else {
-      const allRecords: Vec<any> = await this.publicApi.query.system.events.at(
-        signedBlock.block.header.parentHash
-      )
-      for (const record of allRecords) {
-        const { section, method, data } = record.event
-        if (section === "subspace" && method === "RootBlockStored") {
-          const segmentIndex = data[0].asV0.segmentIndex.toNumber()
-          return segmentIndex <= 1 ? 1 : segmentIndex
-        }
-      }
-      return await this.getNetworkSegmentIndex(
-        signedBlock.block.header.parentHash
-      )
-    }
+  public async getNetworkSegmentIndex(): Promise<number> {
+    const recordsRoot = await this.publicApi.query.subspace.counterForRecordsRoot() as u32
+    return recordsRoot.toNumber()
```

Comment:
This is not index though, this is number of elements

Reply:
Sure, I had to update several references but at the same time, fixed a few weird variable names. looks better this way.